#!/usr/bin/env python

PACKAGE = 'opencv_apps'

from dynamic_reconfigure.parameter_generator_catkin import *;

gen = ParameterGenerator()

gen.add("use_camera_info", bool_t, 0, "Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used.", False)
filter_type = gen.enum([ gen.const("Homogeneous_Blur",   int_t,  0, "Homogeneous blur"),
                         gen.const("Gaussian_Blur",   int_t,  1, "Gaussian blur"),
                         gen.const("Median_Blur",   int_t,  2, "Median blur"),
                         gen.const("Bilateral_Filter",   int_t,  3, "Bilateral Filter")], "An enum for Smoothing Filter Mehtods")

gen.add("filter_type", int_t, 0, "Smoothing Filter Methods", 1, 0, 3, edit_method=filter_type)
gen.add("kernel_size", int_t, 0, "Size of the kernel (only one because we use a square window). Must be odd.", 7, 1, 31)

exit (gen.generate (PACKAGE, "smoothing", "Smoothing"))
